frontend/src/APIClients/BaseGQLClient.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ApolloClient,
+  ApolloLink,
+  createHttpLink,
+  InMemoryCache,
+  Observable,
+} from "@apollo/client";
+import { onError } from "@apollo/client/link/error";
+import { setContext } from "@apollo/client/link/context";
+import { RetryLink } from "@apollo/client/link/retry";
+import jwt from "jsonwebtoken";
+import AUTHENTICATED_USER_KEY from "../constants/AuthConstants";
+import {
+  setLocalStorageObjProperty,
+  getLocalStorageObjProperty,
+} from "../utils/LocalStorageUtils";
+import { REFRESH } from "./mutations";
+
+const getLocalToken = () =>
+  getLocalStorageObjProperty(AUTHENTICATED_USER_KEY, "accessToken");
+
+const httpLink = createHttpLink({
+  uri: `${process.env.REACT_APP_BACKEND_URL}/graphql`,
+  credentials: "include",
+});
+
+const authFromLocalLink = setContext(async (_, { headers }) => {
+  const accessToken = getLocalToken();
+  return {
+    headers: {
+      ...headers,
+      Authorizaton: accessToken ? `Bearer ${accessToken}` : "",
+    },
+  };
+});
+
+const injectAccessToken = async (operation: any) => {
+  let accessToken = getLocalToken();
+  const decodedToken: any = jwt.decode(accessToken);
```

Comment:
does the accessToken object type change?

Reply:
no, i left it in as `any` as im a little lazy about looking into the return type of `jwt.decode(accessToken)`, looks like its not baked into jwt and needs to be self declared
```
interface MyToken {
  name: string;
  exp: number;
  // whatever else is in the JWT.
}

const decodedToken = jwtDecode<MyToken>(token);
console.log(decodedToken.exp); // works!
```
https://stackoverflow.com/questions/61199530/typescript-error-with-accessing-jwt-decode-object